File: dia-02/07-expressoes-aritmeticas/04-exercicio2-corrigido.cs
// 2. Exploração do operador módulo (%) - Desafiador/Opcional

using System;

class App
{
    static void Main()
    {
        Console.Write("Digite um número inteiro de três dígitos: ");
        int numero = int.Parse(Console.ReadLine()); // ou Convert.ToInt32(Console.ReadLine());

        int soma = 0;

        // Unidade
        int unidade = numero % 10;
        Console.WriteLine($"Unidade: {unidade}");
        soma += unidade;

        // Dezena
        int dezena = (numero / 10) % 10;
        Console.WriteLine($"Dezena: {dezena}");
        soma += dezena;

        // Centena
        int centena = (numero / 100) % 10;
        Console.WriteLine($"Centena: {centena}");
        soma += centena;

        Console.WriteLine($"Soma dos dígitos: {soma}");
    }
}

File: MM5 Front/curso-adaptacao-js-edicao-2025/dia-01/04-expressoes-aritmeticas/04-exercicio2-corrigido.js
const numero = 472;
let soma = 0;

const unidade = numero % 10;
console.log("Unidade: " + unidade);
soma += unidade;

const dezena = Math.floor(numero / 10) % 10;
console.log("Dezena: " + dezena);
soma += dezena;

const centena = Math.floor(numero / 100) % 10;
console.log("Centena: " + centena);
soma += centena;

console.log("Soma dos dígitos: " + soma);